src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Document.cs

```diff
@@ -105,6 +105,7 @@ public static partial class JsonSerializer
                 ThrowHelper.ThrowArgumentNullException(nameof(jsonTypeInfo));
             }
 
+            jsonTypeInfo.EnsureConfigured();
```

Comment:
Did we actually hit any issue with these? I'm surprised none of the tests failed. Aren't we configuring in `ReadDocument` or futher?

Reply:
same question for other EnsureConfigured

Reply:
> Did we actually hit any issue with these? I'm surprised none of the tests failed. Aren't we configuring in ReadDocument or futher?

We did, that's what #72630 was fixing. `EnsureConfigured()` will eventually get called but only on an on-demand basis. This ensures that the value is locked as soon as it is passed to the serializer.